map/bookmark.hpp

```diff
@@ -96,7 +97,13 @@ class Bookmark : public ICustomDrawable
   void SetName(string const & name) { m_data.SetName(name); }
   /// @return Now its a bookmark color - name of icon file
   string const & GetType() const { return m_data.GetType(); }
-  void SetType(string const & type) { m_data.SetType(type); }
+
+  void SetType(string const & type)
+  {
+    m_data.SetType(type);
+    SetStyle(type);
```

Comment:
For example, you can declare pure protected virtual method StyledPoint::GetStyle() and implement it in Bookmark and ApiMarkPoint. Boorkmark implementation will return style from m_data and ApiMarkPoint will return ApiMarkPoint::m_style.